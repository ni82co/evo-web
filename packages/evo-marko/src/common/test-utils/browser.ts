import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el: HTMLElement, to: number, cb: Function) {
  fireEvent.scroll(el);
  el.scrollLeft = to;
  setTimeout(cb, 600);
}

function waitFrames(count: number, cb: Function) {
  if (count) {
    return requestAnimationFrame(() => {
      waitFrames(count - 1, cb);
    });
  }

  cb();
}

interface FastAnimations {
  fastAnimationStyle?: HTMLStyleElement;
  start: () => void;
  stop: () => void;
}

const fastAnimations:FastAnimations = {
  // Adds an style to the document which forces all transitions to run more quickly for the tests.
  start() {
    if (this.fastAnimationStyle) {
      return;
    }

    this.fastAnimationStyle = document.createElement("style");
    this.fastAnimationStyle.innerHTML = `* {
                transition-duration: 0.1s !important;
                transition-delay: 0.1s !important;
                scroll-behavior: auto !important;
            }`;
    document.head.appendChild(this.fastAnimationStyle);
  },
  stop() {
    if (!this.fastAnimationStyle) {
      return;
    }

    document.head.removeChild(this.fastAnimationStyle);
    this.fastAnimationStyle = undefined;
  },
};

export { fastAnimations, waitFrames, simulateScroll };
