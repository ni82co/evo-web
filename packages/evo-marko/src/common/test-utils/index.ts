export * from "./browser"
export * from "./snapshots";