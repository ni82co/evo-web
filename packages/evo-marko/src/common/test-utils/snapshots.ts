import { render, normalize } from "@marko/testing-library";
import { expect } from "vitest";

async function snapshotHTML(template: any, input?: any) {
  const { container } = await render(template, input);
  expect(normalize(container)).toMatchSnapshot();
}

export { snapshotHTML };
