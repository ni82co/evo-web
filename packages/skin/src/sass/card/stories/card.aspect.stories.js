export default { title: "Skin/Card/Aspect" };

export const verticalAspectSquare = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>

        <div class="card__footer">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const verticalAspect_5by4 = () => `
<span class="card card--aspect-5-4">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Football rookie cards so pristine that they've never been touched by human hands, only examined through specialized gloves by card whisperers.
        </div>

        <div class="card__footer">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const verticalAspect_9by16 = () => `
<span class="card card--aspect-16-9">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine â€“ stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>

        <div class="card__footer">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;
