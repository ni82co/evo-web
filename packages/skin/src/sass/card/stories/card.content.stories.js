export default { title: "Skin/Card/Content" };

export const minimumContent = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>

        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const maximumContent = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__overline">
            <svg class="icon icon--16" height="16" width="16">
                <use href="#icon-authenticity-guarantee-filled-16-colored" />
            </svg>
            <span class="card__overline-text">Authenticity Guaranteed</span>
        </div>
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            This legendary rookie card collection features future Hall of Famers back when their biggest concern was remembering to smile for the camera.
        </div>

        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const usingAnchor = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>
        <div class="card__action">
            <a href="https://ebay.com" class="fake-btn fake-btn--primary fake-btn--fluid">See Details</a>
        </div>
    </div>
</span>
`;

export const usingButton = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>
        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;
