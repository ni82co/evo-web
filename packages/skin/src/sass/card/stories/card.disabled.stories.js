export default { title: "Skin/Card/Disabled" };

export const internallyActionable = () => `
<span class="card card--aspect-16-9">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine – stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>
        <div class="card__action">
            <button class="btn btn--primary btn--fluid" tabindex="-1" disabled>See Details</button>
        </div>
    </div>
</span>
`;

export const selfActionableButton = () => `
<button class="card card--aspect-16-9" disabled>
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine – stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>
    </div>
</button>
`;

export const selfActionableLink = () => `
<a class="card card--aspect-16-9" href="" tabindex="-1">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine – stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>
    </div>
</a>
`;
