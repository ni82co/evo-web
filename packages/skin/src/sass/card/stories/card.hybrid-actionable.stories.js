export default { title: "Skin/Card/Hybrid Actionable" };

export const vertical = () => `
<div class="card card--outlined">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__overline">
            <svg class="icon icon--16" height="16" width="16">
                <use href="#icon-authenticity-guarantee-filled-16-colored" />
            </svg>
            <span class="card__overline-text">Authenticity Guaranteed</span>
        </div>
        <div class="card__title">Authentic Rookie Cards</div>
        <div class="card__description">
            Rookie cards that serve as a hedge against the apocalypse – when society collapses, surely someone will trade food for a mint Luka Dončić card.
        </div>

        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</div>
`;

export const horizontal = () => `
<div class="card card--horizontal card--outlined">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__overline">
            <svg class="icon icon--16" height="16" width="16">
                <use href="#icon-authenticity-guarantee-filled-16-colored" />
            </svg>
            <span class="card__overline-text">Authenticity Guaranteed</span>
        </div>
        <div class="card__title">Authentic Rookie Cards</div>
        <div class="card__description">
            Rookie cards that serve as a hedge against the apocalypse – when society collapses, surely someone will trade food for a mint Luka Dončić card.
        </div>

        <div class="card__action">
            <button class="btn btn--primary">See Details</button>
        </div>
    </div>
</div>
`;
