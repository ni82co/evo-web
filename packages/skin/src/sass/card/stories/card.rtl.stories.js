export default { title: "Skin/Card/RTL" };

export const vertical = () => `
<div dir="rtl">
    <span class="card">
        <div class="card__media">
            <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
        </div>
        <div class="card__main">
            <div class="card__overline">
                <svg class="icon icon--16" height="16" width="16">
                    <use href="#icon-authenticity-guarantee-filled-16-colored" />
                </svg>
                <span class="card__overline-text">Authenticity Guaranteed</span>
            </div>
            <h3 class="card__title">Authentic Rookie Cards</h3>
            <div class="card__description">
                This legendary rookie card collection features future Hall of Famers back when their biggest concern was remembering to smile for the camera.
            </div>

            <div class="card__footer">
                <button class="btn btn--primary btn--fluid">See Details</button>
            </div>
        </div>
    </span>
</div>
`;

export const horizontal = () => `
<div dir="rtl">
    <span class="card card--horizontal">
        <div class="card__media">
            <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
        </div>
        <div class="card__main">
            <div class="card__overline">
                <svg class="icon icon--16" height="16" width="16">
                    <use href="#icon-authenticity-guarantee-filled-16-colored" />
                </svg>
                <span class="card__overline-text">Authenticity Guaranteed</span>
            </div>
            <h3 class="card__title">Authentic Rookie Cards</h3>
            <div class="card__description">
                This legendary rookie card collection features future Hall of Famers back when their biggest concern was remembering to smile for the camera.
            </div>

            <div class="card__footer">
                <button class="btn btn--primary">See Details</button>
            </div>
        </div>
    </span>
</div>
`;