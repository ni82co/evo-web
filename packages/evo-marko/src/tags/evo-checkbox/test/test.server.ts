import { afterEach, beforeEach, describe, it, expect, vi } from "vitest";
import { render, cleanup, fireEvent } from "@marko/testing-library";
import { composeStories } from "@storybook/marko";
import * as stories from "../checkbox.stories"; // import all stories from the stories file
import { snapshotHTML } from "../../../common/test-utils/snapshots";
const { Group, Disabled, Isolated } = composeStories(stories);

describe("evo-checkbox", () => {
  it("renders default checkbox", async () => {
    await snapshotHTML(Isolated);
  });

  it("renders disabled checkbox", async () => {
    await snapshotHTML(Disabled);
  });

  it("renders checkbox with id", async () => {
    await snapshotHTML(Group);
  });
});
