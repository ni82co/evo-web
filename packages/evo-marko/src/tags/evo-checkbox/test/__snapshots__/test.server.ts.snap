// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`evo-checkbox > renders checkbox with id 1`] = `
<DocumentFragment>
  <div>
    false,false,false
  </div>
  <fieldset>
    <legend>
      Choose an Option
    </legend>
    <span
      class="field"
    >
      <span
        class="checkbox"
      >
        <input
          class="checkbox__control"
          id="GENERATED-0"
          name="checkbox-group"
          type="checkbox"
        />
        <span
          class="checkbox__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="checkbox__unchecked icon icon--18"
          >
            <symbol
              id="GENERATED-1"
              viewBox="0 0 18 18"
            >
              <path
                d="M6.4 17h5.2c1.137 0 1.929 0 2.546-.051.605-.05.953-.142 1.216-.276a3 3 0 0 0 1.311-1.311c.134-.263.226-.611.276-1.216.05-.617.051-1.41.051-2.546V6.4c0-1.137 0-1.929-.051-2.546-.05-.605-.142-.953-.276-1.216a3 3 0 0 0-1.311-1.311c-.263-.134-.611-.226-1.216-.276C13.529 1.001 12.736 1 11.6 1H6.4c-1.137 0-1.929 0-2.546.051-.605.05-.953.142-1.216.276a3 3 0 0 0-1.311 1.311c-.134.263-.226.611-.276 1.216C1.001 4.471 1 5.264 1 6.4v5.2c0 1.137 0 1.929.051 2.546.05.605.142.953.276 1.216a3 3 0 0 0 1.311 1.311c.263.134.611.226 1.216.276.617.05 1.41.051 2.546.051ZM.436 15.816C0 14.96 0 13.84 0 11.6V6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748Z"
              />
            </symbol>
            <use
              href="#icon-checkbox-unchecked-18"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="checkbox__checked icon icon--18"
          >
            <symbol
              id="GENERATED-2"
              viewBox="0 0 18 18"
            >
              <path
                d="M0 6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748C0 14.96 0 13.84 0 11.6V6.4Zm14.71-2.11a1 1 0 0 0-1.41 0L7 10.59l-2.29-2.3a1.004 1.004 0 0 0-1.42 1.42l3 3a1 1 0 0 0 1.41 0l7-7a1 1 0 0 0 .01-1.42Z"
              />
            </symbol>
            <use
              href="#icon-checkbox-checked-18"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="GENERATED-0"
      >
        Option 1
      </label>
    </span>
    <span
      class="field"
    >
      <span
        class="checkbox"
      >
        <input
          class="checkbox__control"
          id="GENERATED-3"
          name="checkbox-group"
          type="checkbox"
        />
        <span
          class="checkbox__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="checkbox__unchecked icon icon--18"
          >
            <use
              href="#icon-checkbox-unchecked-18"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="checkbox__checked icon icon--18"
          >
            <use
              href="#icon-checkbox-checked-18"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="GENERATED-3"
      >
        Option 2
      </label>
    </span>
    <span
      class="field"
    >
      <span
        class="checkbox"
      >
        <input
          class="checkbox__control"
          id="GENERATED-4"
          name="checkbox-group"
          type="checkbox"
        />
        <span
          class="checkbox__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="checkbox__unchecked icon icon--18"
          >
            <use
              href="#icon-checkbox-unchecked-18"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="checkbox__checked icon icon--18"
          >
            <use
              href="#icon-checkbox-checked-18"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="GENERATED-4"
      >
        Option 3
      </label>
    </span>
  </fieldset>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.o=[0,_.b={"LoopScopeMap:#text/1":new Map(_.a=[[0,_.c={i:0,checked:!1,"#childScope/0":_.d={"ControlledType:#input/0":0,"ConditionalRenderer:#span/1":1,"ConditionalScope:#span/1":_.l={},htmlInput:_.e={checked:!1,class:"field__control",id:"group-checkbox-0",value:0,name:"checkbox-group"}}}],[1,_.f={i:1,checked:!1,"#childScope/0":_.g={"ControlledType:#input/0":0,"ConditionalRenderer:#span/1":1,"ConditionalScope:#span/1":_.m={},htmlInput:_.h={checked:!1,class:"field__control",id:"group-checkbox-1",value:1,name:"checkbox-group"}}}],[2,_.i={i:2,checked:!1,"#childScope/0":_.j={"ControlledType:#input/0":0,"ConditionalRenderer:#span/1":1,"ConditionalScope:#span/1":_.n={},htmlInput:_.k={checked:!1,class:"field__control",id:"group-checkbox-2",value:2,name:"checkbox-group"}}}]]),input:{},checkedItems:[!1,!1,!1]},_.c,_.d,_.l,1,{htmlInput:{}},1,{htmlInput:{}},_.f,_.g,_.m,1,{htmlInput:{}},1,{htmlInput:{}},_.i,_.j,_.n,1,{htmlInput:{}},1,{htmlInput:{}}],_.c._=_.f._=_.i._=_.b,_.c["TagVariableChange:checked"]=_._["src/tags/evo-checkbox/examples/group.marko_1/valueChange"](_.c),_.d["ControlledHandler:#input/0"]=_.e.checkedChange=_._["src/tags/evo-checkbox/examples/group.marko_1/checkedChange"](_.c),_.f["TagVariableChange:checked"]=_._["src/tags/evo-checkbox/examples/group.marko_1/valueChange"](_.f),_.g["ControlledHandler:#input/0"]=_.h.checkedChange=_._["src/tags/evo-checkbox/examples/group.marko_1/checkedChange"](_.f),_.i["TagVariableChange:checked"]=_._["src/tags/evo-checkbox/examples/group.marko_1/valueChange"](_.i),_.j["ControlledHandler:#input/0"]=_.k.checkedChange=_._["src/tags/evo-checkbox/examples/group.marko_1/checkedChange"](_.i),_.o),"src/tags/evo-icon/index.marko_0_htmlInput",6,8,"src/tags/evo-checkbox/index.marko_0_htmlInput",3,"src/tags/evo-icon/index.marko_0_htmlInput",13,15,"src/tags/evo-checkbox/index.marko_0_htmlInput",10,"src/tags/evo-icon/index.marko_0_htmlInput",20,22,"src/tags/evo-checkbox/index.marko_0_htmlInput",17];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-checkbox > renders default checkbox 1`] = `
<DocumentFragment>
  <span
    class="checkbox"
  >
    <input
      class="checkbox__control"
      type="checkbox"
    />
    <span
      class="checkbox__icon"
      hidden=""
    >
      <svg
        aria-hidden="true"
        class="checkbox__unchecked icon icon--18"
      >
        <symbol
          id="GENERATED-0"
          viewBox="0 0 18 18"
        >
          <path
            d="M6.4 17h5.2c1.137 0 1.929 0 2.546-.051.605-.05.953-.142 1.216-.276a3 3 0 0 0 1.311-1.311c.134-.263.226-.611.276-1.216.05-.617.051-1.41.051-2.546V6.4c0-1.137 0-1.929-.051-2.546-.05-.605-.142-.953-.276-1.216a3 3 0 0 0-1.311-1.311c-.263-.134-.611-.226-1.216-.276C13.529 1.001 12.736 1 11.6 1H6.4c-1.137 0-1.929 0-2.546.051-.605.05-.953.142-1.216.276a3 3 0 0 0-1.311 1.311c-.134.263-.226.611-.276 1.216C1.001 4.471 1 5.264 1 6.4v5.2c0 1.137 0 1.929.051 2.546.05.605.142.953.276 1.216a3 3 0 0 0 1.311 1.311c.263.134.611.226 1.216.276.617.05 1.41.051 2.546.051ZM.436 15.816C0 14.96 0 13.84 0 11.6V6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748Z"
          />
        </symbol>
        <use
          href="#icon-checkbox-unchecked-18"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="checkbox__checked icon icon--18"
      >
        <symbol
          id="GENERATED-1"
          viewBox="0 0 18 18"
        >
          <path
            d="M0 6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748C0 14.96 0 13.84 0 11.6V6.4Zm14.71-2.11a1 1 0 0 0-1.41 0L7 10.59l-2.29-2.3a1.004 1.004 0 0 0-1.42 1.42l3 3a1 1 0 0 0 1.41 0l7-7a1 1 0 0 0 .01-1.42Z"
          />
        </symbol>
        <use
          href="#icon-checkbox-checked-18"
        />
      </svg>
    </span>
  </span>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.a=[0,{htmlInput:{}},2,{htmlInput:{}},1,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",4,6,"src/tags/evo-checkbox/index.marko_0_htmlInput",1];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-checkbox > renders disabled checkbox 1`] = `
<DocumentFragment>
  <span
    class="field"
  >
    <span
      class="checkbox"
    >
      <input
        class="checkbox__control"
        disabled=""
        id="checkbox"
        type="checkbox"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--18"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 18 18"
          >
            <path
              d="M6.4 17h5.2c1.137 0 1.929 0 2.546-.051.605-.05.953-.142 1.216-.276a3 3 0 0 0 1.311-1.311c.134-.263.226-.611.276-1.216.05-.617.051-1.41.051-2.546V6.4c0-1.137 0-1.929-.051-2.546-.05-.605-.142-.953-.276-1.216a3 3 0 0 0-1.311-1.311c-.263-.134-.611-.226-1.216-.276C13.529 1.001 12.736 1 11.6 1H6.4c-1.137 0-1.929 0-2.546.051-.605.05-.953.142-1.216.276a3 3 0 0 0-1.311 1.311c-.134.263-.226.611-.276 1.216C1.001 4.471 1 5.264 1 6.4v5.2c0 1.137 0 1.929.051 2.546.05.605.142.953.276 1.216a3 3 0 0 0 1.311 1.311c.263.134.611.226 1.216.276.617.05 1.41.051 2.546.051ZM.436 15.816C0 14.96 0 13.84 0 11.6V6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748Z"
            />
          </symbol>
          <use
            href="#icon-checkbox-unchecked-18"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--18"
        >
          <symbol
            id="GENERATED-1"
            viewBox="0 0 18 18"
          >
            <path
              d="M0 6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748C0 14.96 0 13.84 0 11.6V6.4Zm14.71-2.11a1 1 0 0 0-1.41 0L7 10.59l-2.29-2.3a1.004 1.004 0 0 0-1.42 1.42l3 3a1 1 0 0 0 1.41 0l7-7a1 1 0 0 0 .01-1.42Z"
            />
          </symbol>
          <use
            href="#icon-checkbox-checked-18"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end field__label--disabled"
      for="checkbox"
    >
      Option
    </label>
  </span>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.a=[0,1,{htmlInput:{checked:!1,disabled:!0,class:"field__control",id:"checkbox"}},2,{htmlInput:{}},1,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",5,7,"src/tags/evo-checkbox/index.marko_0_htmlInput",2];M._.w()
  </script>
</DocumentFragment>
`;
